src/classlibnative/bcltype/bignum.cpp

```diff
@@ -0,0 +1,599 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// File: bignum.cpp
+//
+
+//
+
+#include "bignum.h"
+#include <intrin.h>
+
+constexpr UINT32 BigNum::m_power10UInt32Table[UINT32POWER10NUM];
+BigNum BigNum::m_power10BigNumTable[BIGPOWER10NUM];
+BigNum::StaticInitializer BigNum::m_initializer;
+
+BigNum::BigNum()
+    :m_len(0) // Note: we do not zeroing m_blocks due to performance.
+{
+}
+
+BigNum::BigNum(UINT32 value)
+{
+    SetUInt32(value);
+}
+
+BigNum::BigNum(UINT64 value)
+{
+    SetUInt64(value);
+}
+
+BigNum::~BigNum()
+{
+}
+
+BigNum& BigNum::operator=(const BigNum &rhs)
+{
+    memcpy(m_blocks, rhs.m_blocks, ((UINT32)rhs.m_len) * sizeof(UINT32));
+    m_len = rhs.m_len;
+
+    return *this;
+}
+
+int BigNum::Compare(const BigNum& lhs, const BigNum& rhs)
+{
+    int lenDiff = (int)lhs.m_len - (int)rhs.m_len;
+    if (lenDiff != 0)
+    {
+        return lenDiff;
+    }
+
+    if (lhs.m_len == 0 && rhs.m_len == 0)
+    {
+        return 0;
+    }
+
+    UINT32 lastIndex = lhs.m_len - 1;
+    INT32 currentIndex = lastIndex;
+
+    while (currentIndex >= 0)
```

Comment:
`for (UINT32 currentIndex = lhs.m_len - 1; currentIndex >= 0; --currentIndex)` might be more readable?